pkg/rc/replication_controller.go

```diff
@@ -194,6 +186,42 @@ func (rc *replicationController) WatchDesires(quit <-chan struct{}) <-chan error
 	// When seeing any changes, try to meet them.
 	// If meeting produces any error, send it on the output error channel.
 	go func() {
+		defer func() {
+			rc.nodeTransferMu.Lock()
+			if rc.nodeTransfer.quit != nil {
+				rc.nodeTransfer.rollbackReason = "RC farm quitting"
+				close(rc.nodeTransfer.quit)
+			}
+			rc.nodeTransferMu.Unlock()
+		}()
+		// Fetch the RC's status to find out of a node transfer is underway
+		// from a different incarnation of this RC
+
+		needToFetchStatus := true
+		var status rcstatus.Status
+		var err error
+		for needToFetchStatus {
+			status, _, err = rc.rcStatusStore.Get(rc.rcID)
+			switch {
+			case err == nil:
+				needToFetchStatus = false
+			case statusstore.IsNoStatus(err):
+				needToFetchStatus = false
+			case err != nil:
+				errOutChannel <- err
+				rc.logger.WithError(err).Errorln("cannot begin servicing RC because status could not be fetched")
+				time.Sleep(1 * time.Second)
```

Comment:
When is it correct/better to use exponential backoff and when does one not bother?

Reply:
we should probably be doing it everywhere in all error scenario, here included, i'll add it